src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Security/AuthorizationHandlers/QueryDataSetVersionAuthorizationHandler.cs

```diff
@@ -32,8 +33,9 @@ protected override async Task HandleRequirementAsync(
     private async Task<bool> RequestHasValidPreviewToken(DataSetVersion dataSetVersion)
     {
         return httpContextAccessor.HttpContext.TryGetRequestHeader(
-                   RequestHeaderConstants.PreviewTokenRequestHeaderName,
-                   out var previewToken)
-               && await previewTokenService.ValidatePreviewToken(previewToken, dataSetVersion.Id);
+                   RequestHeaderNames.PreviewToken,
+                   out var previewTokenValue)
+               && previewTokenValue.Count == 1
+               && await previewTokenService.ValidatePreviewToken(previewTokenValue!, dataSetVersion.Id);
```

Comment:
Don't need the count check? Could do:

```cs
return httpContextAccessor.HttpContext.TryGetRequestHeader(
           RequestHeaderNames.PreviewToken,
           out var previewToken)
       && await previewTokenService.ValidatePreviewToken(previewToken.ToString(), dataSetVersion.Id);
```

The `ToString` will concatenate any preview token header values together and if there are multiple values, this would create an invalid `Guid` so would fail `ValidatePreviewToken`.

Reply:
Good idea, I've made this change in the latest fixup commit. Thanks